File: 8-kyu-kata/returning-strings/python/returning_strings_test.py
# 8 kyu - Returning Strings  [ ID: 55a70521798b14d4750000a4  (returning-strings) ]
# URL: https://www.codewars.com/kata/55a70521798b14d4750000a4
# Category: REFERENCE  |  Tags: STRINGS | FUNDAMENTALS
# ******************************************************************************
import codewars_test as test
from solution import greet

@test.describe("Fixed Tests")
def basic_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(greet('Ryan'), "Hello, Ryan how are you doing today?")
        test.assert_equals(greet('Shingles'), "Hello, Shingles how are you doing today?")

@test.describe("Random Tests")
def random_tests():
    
    from random import randint, choice
    
    def randgen_sent():
        rand = randint(10, 50)
        base = ' abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ!?.,'
        sent = ''.join(choice(base) for i in range(rand))
        return sent 

    actual=lambda s: "Hello, {} how are you doing today?".format(s)

    for _ in range(100):
        s = randgen_sent()
        expected = actual(s)
        @test.it(f'Testing for greet({s})')
        def test_case():
            test.assert_equals(greet(s), expected, "It should work for random inputs too")

File: kata-8-kyu/returning-strings/javascript/returningStrings.Test.js
//+ ====================================================================================================================
//+
//+ 8 kyu - Returning Strings  [ ID: 55a70521798b14d4750000a4 ] (returning-strings)
//+ URL: https://www.codewars.com/kata/55a70521798b14d4750000a4
//+ Category: REFERENCE  |  Tags: STRINGS | FUNDAMENTALS
//+
//+ ====================================================================================================================

import { assert } from "vitest"
const { greet } = require("./returningStrings")

describe("Basic tests", () => {
  it("Testing for fixed tests", () => {
    assert.strictEqual(greet("Ryan"), "Hello, Ryan how are you doing today?")
    assert.strictEqual(greet("Shingles"), "Hello, Shingles how are you doing today?")
  })
})

describe("Random tests", () => {
  const randint = (a, b) => ~~(Math.random() * (b - a + 1)) + a
  const sol = (n) => `Hello, ${n} how are you doing today?`
  const base = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"

  it("Testing for 40 random tests", () => {
    for (let i = 0; i < 40; i++) {
      const s = Array.from({ length: randint(1, 30) }, () => base[randint(0, base.length - 1)]).join("")
      assert.strictEqual(greet(s), sol(s), `Testing for ${JSON.stringify(s)}`)
    }
  })
})
